.github/workflows/ci.yaml

```diff
@@ -101,38 +101,51 @@ jobs:
-        kind delete cluster 2>/dev/null
-        kind create cluster
-        kubectl get nodes
-    - name: Deploy From Source
+        skaffold config set --global local-cluster false
+        skaffold build --default-repo=gcr.io/$PROJECT_ID/$GITHUB_REF \
+                       --tag=$GITHUB_SHA
+      env:
+        PROJECT_ID: ${{ secrets.PROJECT_ID }}
+    - name: Deploy PR images to PR GKE cluster
+      timeout-minutes: 20
       run: |
-        skaffold config set --global local-cluster true
-        skaffold run --default-repo local \
-                     --profile ${{ matrix.profile }}
+        PR_NUMBER=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+        NAMESPACE="pr${PR_NUMBER}"
+        gcloud container clusters get-credentials $PR_CLUSTER --zone $ZONE --project $PROJECT_ID
+        kubectl create namespace $NAMESPACE
+        skaffold config set --global local-cluster false
+        skaffold run --default-repo=gcr.io/$PROJECT_ID/$GITHUB_REF --tag=$GITHUB_SHA --namespace=$NAMESPACE
+      env:
+        PROJECT_ID: ${{ secrets.PROJECT_ID }}
+        PR_CLUSTER: ${{ secrets.PR_CLUSTER_1 }}
+        ZONE: ${{ secrets.ZONE }}
     - name: Wait For Pods
       timeout-minutes: 20
       run: |
         set -x
-        kubectl wait --for=condition=available --timeout=500s deployment/frontend
-        kubectl wait --for=condition=available --timeout=500s deployment/contacts
-        kubectl wait --for=condition=available --timeout=500s deployment/userservice
-        kubectl wait --for=condition=available --timeout=500s deployment/ledgerwriter
-        kubectl wait --for=condition=available --timeout=500s deployment/transactionhistory
-        kubectl wait --for=condition=available --timeout=500s deployment/balancereader
-        kubectl wait --for=condition=available --timeout=500s deployment/loadgenerator
-        kubectl wait --for=condition=ready --timeout=500s pod/accounts-db-0
-        kubectl wait --for=condition=ready --timeout=500s pod/ledger-db-0
+        PR_NUMBER=$(echo $GITHUB_REF | awk 'BEGIN { FS = "/" } ; { print $3 }')
+        NAMESPACE="pr${PR_NUMBER}"
+        kubectl config set-context --current --namespace=$NAMESPACE
+        kubectl wait --for=condition=available --timeout=300s deployment/frontend
+        kubectl wait --for=condition=available --timeout=300s deployment/contacts
+        kubectl wait --for=condition=available --timeout=300s deployment/userservice
+        kubectl wait --for=condition=available --timeout=300s deployment/ledgerwriter
+        kubectl wait --for=condition=available --timeout=300s deployment/transactionhistory
+        kubectl wait --for=condition=available --timeout=300s deployment/balancereader
+        kubectl wait --for=condition=available --timeout=300s deployment/loadgenerator
+        kubectl wait --for=condition=ready --timeout=300s pod/accounts-db-0
+        kubectl wait --for=condition=ready --timeout=300s pod/ledger-db-0
     - name: Port Forward frontend
```

Comment:
Do we still need to port forward the frontend?  Now that we're on a GKE cluster, can we use the live IP address that is exposed by the ingress instead?  Let me know if you think that's out of scope for this PR and I can open an issue

Reply:
Will leave for now / to update in @WilliamG240's staging work

Reply:
I think this is in scope, updating!